src/main/java/mil/nga/tiff/Rasters.java

```diff
@@ -36,19 +60,26 @@
 	private final int height;
 
 	/**
-	 * Samples per pixel
+	 * Sample type for each sample.
+	 */
+	private final SampleType[] sampleTypes;
+
+	// Calculated values
+
+	/**
+	 * Calculated pixel size in bytes
 	 */
-	private final int samplesPerPixel;
+	private Integer pixelSize;
 
 	/**
-	 * Bits per sample
+	 * @see getBitsPerSample()
```

Comment:
For my taste, I would omit a cache for bitsPerSample and sampleFormat because (at least in my test program) the methods are called only once or could be stored in a variable externally if someone needs to access it more often.

Reply:
In library's test they're called more often. See TiffTestUtils.java.